### DISCUSSION TITLE
annotating ATAC-seq result for drosophila

### DISCUSSION THREAD

**Discussion by bxxu** [stuart-lab/signac#1088] [Q&A]
Hi, 
I am trying to follow the vignettes for signac to analyze a single cell atac-seq result for drosophila. The seurat object is built like this:
`metadata <- get_cell_annotation(loom)`
`chrom_assay <- CreateChromatinAssay(counts = atacMat, genome = "dm6", min.cells = 10, min.features = 200, sep = c(":","-"))`
`flybrain <- CreateSeuratObject(counts = chrom_assay, assay = "peaks", meta.data = metadata)`

Since there isn't an EnsDb package available for drosophila, I pulled one from annotationhub using this code:
`ah <- AnnotationHub()`
`query(ah, "EnsDb")`
`ahDb <- query(ah, pattern = c("Drosophila", "EnsDb"))`
`flygenome <- ahDb[[19]]`

But when I try to annotate my seurat object following the tutorial:
`annotations <- GetGRangesFromEnsDb(ensdb = flygenome)`
`seqlevelsStyle(annotations) <- 'UCSC'`
`Annotation(flybrain) <- annotations`

It gives this error: Error in SetAssayData.ChromatinAssay(object = object, slot = "annotation",  : 
  Annotation genome does not match genome of the object

What should I do?

Thanks in advance!

**Comment by timoast** [comment]
When you run `genome(annotations)`, it returns `BDGP6.32` not `dm6`. Either set `genome="BDGP6.32"` or remove the `genome` parameter

**Reply by bxxu** [reply]
Thank you so much, that worked! But then I got error when running NucleosoeSignal, TSSEnrichment and GeneActivity. 

I am trying to get to a gene activity/cell matrix from a peak/cell matrix. The peak/cell matrix was downloaded from [https://flybrain.aertslab.org/](url) along with the fragments.tsv and fragments.tsv.gz.tbi files. 

When I run NucleosomeSignal, `flybrain <- NucleosomeSignal(object = flybrain)`, I get : 
Found 31001 cell barcodes
Done Processing 39 million linesFound 31001 cell barcodes
Done Processing 36 million linesFound 31001 cell barcodes
Done Processing 29 million linesFound 31001 cell barcodes
Done Processing 28 million linesFound 31001 cell barcodes
Done Processing 56 million linesError in ecdf(x = af$nucleosome_signal) : 
  'x' must have 1 or more non-missing values

When I run TSSEnrichment, I get:
Extracting TSS positions
Finding + strand cut sites
Finding - strand cut sites
Error in `colnames<-`(`*tmp*`, value = seq_len(length.out = region.width) -  : 
  attempt to set 'colnames' on an object with less than two dimensions

And when I run GeneActivity, I get:
Extracting gene coordinates
Error in SingleFeatureMatrix(fragment = fragments[[x]], features = features,  : 
  No matching chromosomes found in fragment file.

Is the problem associated with the fragment files?

Thanks in advance!

**Reply by timoast** [reply]
It looks like the chromosome names in the fragment file are different to the names in the gene annotation you're using. You'll need to reformat the gene annotation so that they match

**Reply by bxxu** [reply]
Thank you so much!
So I changed the annotation of the genome as from UCSC to NCBI:
`annotations <- GetGRangesFromEnsDb(ensdb = flygenome)`
`seqlevelsStyle(annotations) <- 'NCBI'`

I can now run GeneActivity successfully (though it returns a zero matrix), and it gave the following warnings:
`[W::hts_idx_load2] The index file is older than the data file: G:\braun_lab\BRNAT_ATAC_FRAG\DFB__c0f875__10x_scATAC_Mix_DGRP_pupal_brain_72_APF\fragments.tsv.gz.tbi`

TSSEnrichment can now proceed a bit further, giving a slightly different error:
Extracting TSS positions
Finding + strand cut sites
Finding - strand cut sites
Computing mean insertion frequency in flanking regions
Normalizing TSS score
Error in ecdf(x = object$TSS.enrichment) : 
  'x' must have 1 or more non-missing values

but I still get the exact same error when running NucleosomeSignal.  

Could the fragment file be the problem?

Thanks in advance!

**Reply by timoast** [reply]
Probably the chromosome names in the annotation still don't match the fragment file, you should check the names in both and make sure they match
